functions/consumer/kafka-publisher/src/main/java/org/springframework/cloud/fn/consumer/kafka/KafkaPublisherConfiguration.java

```diff
@@ -0,0 +1,121 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.consumer.kafka;
+
+import java.time.Duration;
+import java.util.function.Consumer;
+
+import org.springframework.boot.autoconfigure.AutoConfiguration;
+import org.springframework.boot.autoconfigure.kafka.KafkaAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.context.properties.PropertyMapper;
+import org.springframework.cloud.fn.common.config.ComponentCustomizer;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.channel.PublishSubscribeChannel;
+import org.springframework.integration.expression.ValueExpression;
+import org.springframework.integration.kafka.dsl.Kafka;
+import org.springframework.integration.kafka.dsl.KafkaProducerMessageHandlerSpec;
+import org.springframework.integration.kafka.outbound.KafkaProducerMessageHandler;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.DefaultKafkaHeaderMapper;
+import org.springframework.lang.Nullable;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+
+/**
+ * A configuration for Apache Kafka Publisher (Consumer function).
+ * Uses a {@link KafkaProducerMessageHandlerSpec} to publish a message to Kafka topic.
+ *
+ * @author Artem Bilan
+ *
+ * @since 4.0
+ */
+@AutoConfiguration(after = KafkaAutoConfiguration.class)
+@EnableConfigurationProperties(KafkaPublisherProperties.class)
+public class KafkaPublisherConfiguration {
+
+	/**
+	 * The function to produce messages to the Kafka topic.
+	 * @param kafkaProducerMessageHandler the handler to publish messages to Kafka.
+	 * @return the consumer for accepting message for producing to Kafka.
+	 */
+	@Bean
+	public Consumer<Message<?>> kafkaPublisher(KafkaProducerMessageHandler<?, ?> kafkaProducerMessageHandler) {
+		return kafkaProducerMessageHandler::handleMessage;
+	}
+
+	@Bean
+	public KafkaProducerMessageHandler<?, ?> kafkaProducerMessageHandlerSpec(KafkaTemplate<?, ?> kafkaTemplate,
+			KafkaPublisherProperties kafkaPublisherProperties,
+			PublishSubscribeChannel kafkaPublisherSuccessChannel,
+			PublishSubscribeChannel kafkaPublisherFailureChannel,
+			PublishSubscribeChannel kafkaPublisherFuturesChannel,
+			@Nullable ComponentCustomizer<KafkaProducerMessageHandlerSpec<?, ?, ?>> kafkaProducerSpecComponentCustomizer) {
+
+		var kafkaProducerMessageHandlerSpec = Kafka.outboundChannelAdapter(kafkaTemplate);
```

Comment:
I notice you start to use `var` here but then do not use it in below or in the tests? Intentional? I have switched to use it everywhere I can and it has made my DX a bit better.

Reply:
I don't know what is DX, but I use `var` only in those cases where the line for variable definition is too long 😄

Reply:
Ahh "DX == developer experience" (ie it makes me a bit happier when I use this, my developer experience increases). 

I have found that `var` does clean up the code a bit.

Reply:
Well, I use whatever my IDEA generates for me when I type `.var` suffix after expression.
And put `var` instead of type, only of it is log.
If my IDEA will generate `var` after that `.var`, I'll be OK with the result 😸